lib/compiler.js

```diff
@@ -402,6 +402,14 @@ function buildMiddlewareInstructions(rootDir, config) {
       allConfigs.forEach(function(config) {
         var resolved = resolveMiddlewarePath(rootDir, middleware);
 
+        // prioritize coffee scripts over json
+        if (!isPreferredExtension(resolved.sourceFile)) {
+          var otherFile = tryOtherExtensions(resolved.sourceFile);
+          if (otherFile !== undefined) {
+            resolved.sourceFile = otherFile;
+          }
```

Comment:
This block is repeated in two places, please extract a new method.

```js
resolved.sourceFile = fixFileExtension(resolved.sourceFile);

function fixFileFileExtension(filepath) {
  if (isPreferredExtension(filepath)) return filepath;
  var otherFile = tryOtherExtensions(filepath);
  return otherFile !== undefined ? otherFile: filepath;
}
```

It may be possible to get rid of `tryOtherExtension` completely and inline it inside the new method, I'll leave that for you to decide.